server/src/internalClusterTest/java/org/opensearch/indices/IndicesRequestCacheIT.java

```diff
@@ -69,6 +75,7 @@
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.greaterThan;
 
+@OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.TEST, numDataNodes = 0, supportsDedicatedMasters = false)
```

Comment:
Considering we are setting nodes to 0, how will the existing tests work? Do we need to startNode for each of them now?

Reply:
okay, after a lot of debugging I understood that when you set numDataNodes = 0 and do not startNode(), it returns a client of `cluster()` which returns a client of `TestCluster` which is a test Cluster for the entire test suite 

However, If you startNode() it would create a test cluster only for the specific test, which would return a client of 
`InternalTestCluster`

```java
Client clientForNode(@Nullable String node) {
        if (node != null) {
            return internalCluster().orElseThrow(() -> new UnsupportedOperationException("current test cluster is immutable")).client(node);
        }
        Client client = cluster().client();
        if (OpenSearchTestCase.frequently()) {
            client = new RandomizingClient(client, OpenSearchTestCase.random());
        }
        return client;
    }
```